From: Jan Blunck <jblunck@suse.de>
Subject: Do the symbolic links right in the first place

Since brp-symlink relinks symbolic links to enforce a certain policy we should
do it right in the first place. So this patch changes find-debuginfo.sh scripts
behavior to reflect that policy.

Signed-off-by: Jan Blunck <jblunck@suse.de>

--- ./scripts/find-debuginfo.sh.orig	2017-12-01 15:26:21.939199791 +0000
+++ ./scripts/find-debuginfo.sh	2017-12-01 15:27:03.153081225 +0000
@@ -305,7 +305,17 @@ debug_link()
   local l="/usr/lib/debug$2"
   local t="$1"
   echo >> "$LINKSFILE" "$l $t"
-  link_relative "$t" "$l" "$RPM_BUILD_ROOT"
+
+  # this should correspond to what brp-symlink is doing
+  case $t in
+      /usr*)
+	  link_relative "$t" "$l" "$RPM_BUILD_ROOT"
+	  ;;
+      *)
+	  mkdir -p "$(dirname "$RPM_BUILD_ROOT$l")" && \
+	      ln -snf "$t" "$RPM_BUILD_ROOT$l"
+	  ;;
+  esac
 }
 
 get_debugfn()
